src/main/java/ml/shifu/shifu/core/dtrain/wdl/WDLParallelGradient.java

```diff
@@ -195,8 +203,8 @@ public WDLParams doCompute() {
      */
     public WDLParams doCompute(int iteration, int miniBatchSize) {
         LOG.info("training on iteration: " + iteration + " with miniBatchSize: " + miniBatchSize);
-        adjustTrainSet(iteration * miniBatchSize % this.trainData.size(), miniBatchSize);
-        return doCompute();
+        adjustTrainSet(miniBatchSize);
```

Comment:
I also don't know why we don't need `start = iteration * miniBatchSize % this.trainData.size()`.

Reply:
We don't need to calculate starting index, because when mini batch enabled, data will be shuffled, and thus effective train data set is already different for each iteration.